src/components/features/occasion-filter/occasion-filter.tsx

```
@@ -0,0 +1,71 @@
+import { getAllOccasions } from "@lib/apis/occasions-filter.api";
+import { useFilters } from "@/hooks/products-filters/use-filters";
+import { useQuery } from "@tanstack/react-query";
+import { Occasion } from "@lib/types/occasions";
+import { useState } from "react";
+
+
+export default function OccasionFilter() {
+  
+  const t = useTranslations("Occasions");
+  const { updateParam, searchParams} = useFilters();
+  const [loading, setLoading] = useState(false);
```

ESLint

[error] 'setLoading' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)